.github/workflows/publish.yml

```diff
@@ -0,0 +1,90 @@
+name: Build and Publish Package
+
+on:
+  create
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref == 'refs/heads/master' && github.run_number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  publish_apisix:
+    name: Build and Publish RPM Package
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v2.3.5
+        with:
+          submodules: recursive
+
+      - name: Extract Tags name
+        if: ${{ startsWith(github.ref, 'refs/tags/') }}
+        id: tag_env
+        shell: bash
+        run: |
+          echo "##[set-output name=version;]$(echo ${GITHUB_REF##*/})"
+
+      - name: Clone apisix-build-tools
+        run: |
+          git clone https://github.com/api7/apisix-build-tools.git
+
+      - name: Build apisix RPM Package
+        if: ${{ startsWith(github.ref, 'refs/tags/') }}
+        run: |
+          cd ./apisix-build-tools
+          # build apisix
+          make package type=rpm app=apisix checkout=${{ steps.tag_env.outputs.version }} version=${{ steps.tag_env.outputs.version }} image_base=centos image_tag=7
+          cd ../
+          mv ./apisix-build-tools/output/apisix-${{ steps.tag_env.outputs.version }}-0.el7.x86_64.rpm ./
+
+      - name: Build apisix-base RPM Package
+        if: ${{ startsWith(github.ref, 'refs/tags/') }}
+        run: |
+          cd ./apisix-build-tools
+          # build apisix-base
+          make package type=rpm app=apisix-base checkout=${{ steps.tag_env.outputs.version }} version=${{ steps.tag_env.outputs.version }} image_base=centos image_tag=7
+          cd ../
+          mv ./apisix-build-tools/output/apisix-base-${{ steps.tag_env.outputs.version }}-0.el7.x86_64.rpm ./
+
+      - name: Upload apisix Artifact
+        if: ${{ startsWith(github.ref, 'refs/tags/') }}
+        uses: actions/upload-artifact@v2.2.3
+        with:
+          name: "apisix-${{ steps.tag_env.outputs.version }}-0.el7.x86_64.rpm"
+          path: "./apisix-${{ steps.tag_env.outputs.version }}-0.el7.x86_64.rpm"
+
+      - name: Upload apisix-base Artifact
+        if: ${{ startsWith(github.ref, 'refs/tags/') }}
+        uses: actions/upload-artifact@v2.2.3
+        with:
+          name: "apisix-base-${{ steps.tag_env.outputs.version }}-0.el7.x86_64.rpm"
+          path: "./apisix-base-${{ steps.tag_env.outputs.version }}-0.el7.x86_64.rpm"
+
+      - name: Push apisix RPM Package to Aliyun OSS
+        if: ${{ startsWith(github.ref, 'refs/tags/') }}
+        run: |
+          if echo $GITHUB_REF | grep "tags"; then
```

Comment:
We already check it in `if: ${{ startsWith(github.ref, 'refs/tags/') }}`?

Reply:
Done. Thanks